doc/source/security/encrypt-traffic.rst

```diff
@@ -4,291 +4,231 @@
+files shoud be specified in the |PXC| configuration. If you do not have the
+necessary files, see :ref:`generate-keys-certs`.
 
 .. note:: Encryption settings are not dynamic.
    To enable it on a running cluster, you need to restart the entire cluster.
 
 There are three aspects of |PXC| operation, where you can enable encryption:
 
-* :ref:`encrypt-client-server`
-
-  This refers to communication between client applications and cluster nodes.
-
-* :ref:`encrypt-replication`
-
-  This refers to all internal |PXC| communication,
-  such as, write-set replication, :term:`IST`, and various service messages.
-
 * :ref:`encrypt-sst`
 
   This refers to :term:`SST` traffic during full data copy
   from one cluster node (donor) to the joining node (joiner).
 
-.. _encrypt-client-server:
-
-Encrypting Database Traffic
----------------------------
-
-|PXC| uses the underlying MySQL encryption mechanism
-to secure communication between client applications and cluster nodes.
-Specify the following settings in the :file:`my.cnf` configuration file
-for each node::
-
- [mysqld]
- ssl-ca=/etc/mysql/certs/ca.pem
- ssl-cert=/etc/mysql/certs/server-cert.pem
- ssl-key=/etc/mysql/certs/server-key.pem
-
- [client]
- ssl-ca=/etc/mysql/certs/ca.pem
- ssl-cert=/etc/mysql/certs/client-cert.pem
- ssl-key=/etc/mysql/certs/client-key.pem
-
-After you restart the node,
-it will use these files for encrypting communication with clients.
-MySQL clients require only the second part of the configuration
-to communicate with cluster nodes.
-
-.. _encrypt-replication:
-
-Encrypting Replication Traffic
-------------------------------
-
-Replication traffic refers to the following:
-
-* Write-set replication is the main workload of |PXC|
-  (replicating transactions that execute on one node to all other nodes).
-* Incremental State Transfer (:term:`IST`)
-  is copying only missing transactions from DONOR to JOINER node.
-* Service messages ensure that all nodes are synchronized.
-
-All of this traffic is transferred
-via the same underlying communication channel (``gcomm``).
-Securing this channel will ensure that IST traffic, write-set replication,
-and service messages are encypted.
-
-To enable encryption for all these processes,
-define the paths to the key, certificate and certificate authority files
-using the following :ref:`wsrep provider options <wsrep_provider_index>`:
-
-* :variable:`socket.ssl_ca`
-* :variable:`socket.ssl_cert`
-* :variable:`socket.ssl_key`
-
-To set these options, use the :variable:`wsrep_provider_options` variable
-in the configuration file::
-
- wsrep_provider_options="socket.ssl=yes;socket.ssl_ca=/etc/mysql/certs/ca.pem;socket.ssl_cert=/etc/mysql/certs/server-cert.pem;socket.ssl_key=/etc/mysql/certs/server-key.pem"
-
-.. note:: You must use the same key and certificate files on all nodes,
-   preferably those used for :ref:`encrypt-client-server`.
+* :ref:`encrypt-replication`
```

Comment:
* :ref:`encrypt-replication`
* :ref:`encrypt-ist`
(We are explicitly pointing out IST as a different mode of traffic so we need to ensure that both the heading are mentioned).